### PR TITLE
Add recast provider to Botpress NLU

### PR DESCRIPTION
**Pull request by ofelt**
Add recast support for intent classification and entity extraction

### CODE REVIEW

File: modules/botpress-nlu/src/providers/recast.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+  /*******
+  Public API
+  *******/
+
+  async init() {
+  }
+
+  async sync() {
+    throw new Error('Not implemented')
```
Comment (slvnperron) [inline modules/botpress-nlu/src/providers/recast.js]: Any ETA for this implementation :) ?

**Review by slvnperron — APPROVED** [review]
Really cool, thanks a bunch for this PR! 

I'd like to get the sync implemented before we can add new providers, any idea of when this will be done?

Great work!

**Comment by ofelt** [comment]
I think PR without the sync should be accepted because one (either the developer or the user) may not be using the sync functionality so it's nice for people to still be able to use the parse feature while waiting for the sync functionality, but I also understand the need to have a complete state for a provider ^^

The synchronization for Recast is done and working on my side I just need to make it "PR ready', so maybe today or tomorrow :)

**Comment by slvnperron** [comment]
@ofelt Great, let me know once that's ready :)

**Comment by ofelt** [comment]
@slvnperron It is ^^ I pushed another commit in the meantime :p

**Review by slvnperron — APPROVED** [review]
That's great work! Thanks Oli :)

**Pull request merged**